riscv/execute.cc

```diff
@@ -173,6 +175,13 @@ static inline reg_t execute_insn_logged(processor_t* p, reg_t pc, insn_fetch_t f
 
   try {
     npc = fetch.func(p, fetch.insn, pc);
+
+    if (p->get_log_commits_enabled()) {
+      hart_to_encoder_ingress_t packet;
```

Comment:
Can you use [designated initializers](https://gcc.gnu.org/onlinedocs/gcc/Designated-Inits.html) here?

Reply:
`hart_to_encoder_ingress_init` contains some more lines of logic than just designated initialization and I think it probably deserves a separate function.

Reply:
You can flatten that entire method into this block. This is the only place its called, and its not doing much else besides assigning the fields of the struct. 
You don't even have to use designated initializers, just inlining it would make this more readable.